src/compiler/proposals-and-stabilization/custom-test-jobs.md

```diff
@@ -0,0 +1,157 @@
+Full CI (the "CI") to fail, or produce a failure message, that may impact other [rust-lang/rust]
+contributors who also use the [rust-lang/rust] CI.
+
+For example (non-exhaustive):
+
+- Ecosystem test jobs: Rust for Linux or Fuchsia.
+- Integration test components: GCC codegen backend or Cranelift codegen backend.
+
+## Background
+
+[rust-lang/rust] runs a small set of build/test jobs on PR CI (faster, less expensive/exhaustive),
+and runs a much larger set of build/test jobs on Full CI. PR CI usually takes around an hour, while
+Full CI usually takes around 3 hours. Having ecosystem and integration test jobs/components that:
+
+- may spuriously fail; or
+- may genuinely fail but it's not clear who should be consulted about the failure or who is
+  responsible for fixing the failure; or
+- otherwise do not have well documented test job maintainers and failure protocol;
+
+can introduce a lot of friction and frustration for other contributors who now will have to deal
+with the failure or understand why the test job failed in the first place, which may now be blocking
+their PR. Everyone utilizing the [rust-lang/rust] CI is expected to use it responsibly.
+
+To help with this, please follow the process described below if you would like to propose adding an
+ecosystem/integration test job/component to the [rust-lang/rust] CI.
+
+## Process for adding an ecosystem/integration test job/component to rust-lang/rust CI
+
+![Flowchart of the process for adding an ecosystem/integration test job/component to rust-lang/rust
+CI](./custom-test-jobs/test-job-flow.svg)
+
+1. Ask the Infrastructure Team on zulip
+  ([#t-infra](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra)) to check if there
+  would be capacity for the proposed test job/component.
+1. Propose using a [Major Change Proposal
+   (MCP)](../proposals-and-stabilization.md#how-do-i-submit-an-mcp).
+    - Including the filled out *Ecosystem and Integration Test Job/Component Policy* (see below).
+1. Once the MCP is seconded and accepted, create a new issue on [rust-lang/rust] about the proposal,
+   linking to the MCP, and then nominate the proposal for library team review via `@rustbot label
+   +I-libs-nominated`.
+    - Link to the MCP.
+1. Once the library team reviews the proposal and has no blocking concerns, submit the
+   implementation PR to [rust-lang/rust].
+    - The PR should include an accompanying Ecosystem/Integration Test Job/Component support page in
+      the in-tree [rustc-dev-guide], ensuring that the *Failure Protocol* is well-documented (see
+      below). Link to the MCP and issue.
+1. The Infrastructure Team will review the implementation PR. Once approved, the
+   ecosystem/integration test job/component will run in [rust-lang/rust] PR CI or Full Merge CI.
+
+## Ecosystem and Integration Test Job/Component Policy
+
+*Please copy and fill out this template as part of the MCP.*
```

Comment:
FIXME(jieyouxu): make it clearer what is part of the template vs what the person is expected to replace and fill out

Reply:
Fixed by a using quoteblocks for policy content itself, italicized placeholder text for things the MCP author is expected to include.